src/t/d8.clj

```diff
@@ -0,0 +1,118 @@
+(ns t.d8
+  (:gen-class)
+  (:require
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]))
+
+(def sample
+"............
+........0...
+.....0......
+.......0....
+....0.......
+......A.....
+............
+............
+........A...
+.........A..
+............
+............")
+
+(defn parse-input
+  [input]
+  (let [lines (cljstr/split-lines input)
+        ymax (dec (count lines))
+        xmax (dec (count (first lines)))]
+  {:xmax xmax
+   :ymax ymax
+   :antennas
+   (->> lines
+        (map-indexed
+          (fn [y l] (map-indexed
+                      (fn [x c] (when (not= c \.) {c [[x y]]}))
+                      l)))
+        (reduce concat)
+        (filter identity)
+        (apply merge-with into)
+        )}))
+
+
+(defn inside?
+  [[x y] xmax ymax]
+  (and (<= 0 x xmax)
+       (<= 0 y ymax)))
+
+(defn antinode
+  [[x1 y1] [x2 y2]]
+  [[(- (* x1 2) x2) (- (* y1 2) y2)]
+   [(- (* x2 2) x1) (- (* y2 2) y1)]])
+
+(defn one-type-antinodes
+  [antennas]
+  (for [a1 antennas
+        a2 antennas
+        :when (not= a1 a2)]
+    (antinode a1 a2)))
+
+(defn d8p1
+  [input]
+  (let [{:keys [xmax ymax antennas]} (parse-input input)]
+    (->>
+      (for [ants (vals antennas)]
+        (->>
+          (one-type-antinodes ants)
+          (reduce concat [])
+          set
```

Comment:
Would need to test but I wonder if perhaps
```
(->> (vals antennas)
      (mapcat one-type-antinodes)
      set
```
would yield the same result here? 🤔

Reply:
probably. I don't know why, but once I started with `for`-s I stopped thinking about `map` (it's bad, I am clearly as lost as last year >.>)